testprojects/src/python/python_distribution/setup_requires/BUILD

```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+python_dist(
+  name='checksummed_version_dist',
+  setup_requires=[
+    # NB: This particular requirement is selected because it is unlikely to already be available to
+    # the python interpreter.
+    'testprojects/3rdparty/python:checksumdir',
+  ],
+)
+
+# NB: The default `sources` for this target is 'test_*.py', so its default sources don't intersect
+# with the `python_dist()`'s default sources.
+python_tests(
+  name='tests',
+  dependencies=[
+    ':checksummed_version_dist',
+  ],
+)
```

Comment:
You'll need to modify `testprojects/src/python/BUILD` so that `:python_distribution_directory` depends on `testprojects/3rdparty:python_directory` (if it doesn't already). Testprojects now run with V2 :)

Reply:
It looks like it does, but could you explain why that's necessary?
```python
files(
  name = 'python_distribution_directory',
  sources = rglobs('python_distribution/*'),
  dependencies = [
    'testprojects/3rdparty:python_directory',
  ],
)
```

Reply:
Great! It’s because the BUILD for at least one of the files within that directory depends on a requirement specified in `testprojects/3rdparty/python`. To make sure that can get included in the chroot, we must declare a dependency on the folder here.

Reply:
Ah ok, thank you!